src/cnaas_nms/api/app.py

```diff
@@ -52,31 +61,50 @@
     }
 }
 
-jwt_query_r = re.compile(r"jwt=[^ &]+")
+jwt_query_r = re.compile(r"code=[^ &]+")
 
 
 class CnaasApi(Api):
     def handle_error(self, e):
         if isinstance(e, DecodeError):
             data = {"status": "error", "data": "Could not decode JWT token"}
+        elif isinstance(e, InvalidKeyError):
+            data = {"status": "error", "data": "Invalid keys {}".format(e)}
         elif isinstance(e, InvalidTokenError):
             data = {"status": "error", "data": "Invalid authentication header: {}".format(e)}
         elif isinstance(e, InvalidSignatureError):
             data = {"status": "error", "data": "Invalid token signature"}
         elif isinstance(e, IndexError):
-            # We might catch IndexErrors which are not cuased by JWT,
+            # We might catch IndexErrors which are not caused by JWT,
             # but this is better than nothing.
             data = {"status": "error", "data": "JWT token missing?"}
         elif isinstance(e, NoAuthorizationError):
             data = {"status": "error", "data": "JWT token missing?"}
         elif isinstance(e, InvalidHeaderError):
             data = {"status": "error", "data": "Invalid header, JWT token missing? {}".format(e)}
+        elif isinstance(e, ExpiredSignatureError):
+            data = {"status": "error", "data": "The JWT token is expired"}
+        elif isinstance(e, MissingAuthorizationError):
+            data = {"status": "error", "data": "JWT token missing?"}
         else:
             return super(CnaasApi, self).handle_error(e)
         return jsonify(data), 401
 
 
 app = Flask(__name__)
+
+# To register the OAuth client
+oauth = OAuth(app)
+client = oauth.register(
+    "connext",
+    server_metadata_url=auth_settings.OIDC_CONF_WELL_KNOWN_URL,
+    client_id=auth_settings.OIDC_CLIENT_ID,
+    client_secret=auth_settings.OIDC_CLIENT_SECRET,
+    client_kwargs={"scope": "openid"},
```

Comment:
For our satosa server to work with this we seem to need to modify this from just "openid" to "openid email" so we can get the email for logging etc later

Reply:
Great, I've added it to the app_setting, so it can  be  set as an env variable or  in the auth_config.yaml

Reply:
Ah great! Can we add something about the new config options on this documentation page also? docs/configuration/index.rst